book/HAS/hybrid-application-service-design.md

```diff
@@ -0,0 +1,88 @@
+# Hybrid Application Service (HAS) Design
+
+## Application CR Create
+
+HAC collects the information required for creating applications. An application is a collection of one or more components. Required information to be collected can be found in the API [doc](). When all the application information is available, HAC will create the Application CR to trigger the application creation. The HAS Application CR Operator will:
+1. Validate all the required fields in the Application CR
+2. If the app model repository and the GitOps repository repos are specified:
+   1. It will validate if the specified repo exists.
+3. If any of those are not specified:
+   1. it will generate a repo under the org https://github.com/redhat-appstudio-appdata
+   2. Update the application CR with the generated repo information
+   3. Triggers the creation of the GitOps repository for the generated repo
+   4. Create the application level devfile under the Application CR status
+
+## Component CR Create
+
+### Importing a Devfile Sample
+HAC will prompt the user to select a sample (retrieved via the Devfile Registry REST API) to choose from. The Git repository corresponding to this sample (such as https://github.com/devfile-samples/devfile-sample-code-with-quarkus) will be specified when creating the Component CR.
+
+### Importing from Git
+HAC will prompt the user to provide a Git repository.
+A ComponentDetectionQuery CR will be created, it will:
+1. If a devfile is present, then:
+   1. If the devfile is valid and has the outerloop Kubernetes component and the deploy command, the information in the Devfile will be returned
+   2. If the Devfile is not valid, the ComponentDetectionQuery will check for a matching devfile from the devfile registry
+2. If the repo doesn’t have a devfile but has a dockerfile, a devfile will be generated to refer to the Dockerfile and store that as part of the component CR.
+3. If a devfile and Dockerfile is not available, the ComponentDetectionQuery will check for a matching devfile from the devfile registry. The ComponentDetectionQuery status.data will be updated to reflect the mapping and a Component stub will be generated. This can be used to create a Component CR to create a component.
+
+If the Git repository contains more than one component, the ComponentDetectionQuery will return a list of components and their associated mapping. If a component is not mapped, then HAC will prompt the user for further information ie, should the directory be ignored, or to manually provide a mapping
```

Comment:
- What are the components mapped to?  
- If this is the first time we are using HAC, let's provide the full name before the acronym